File: drs/src/core/com/db4o/drs/inside/ReplicationPlatform.java
package com.db4o.drs.inside;

import java.util.*;

/**
 * Platform dependent code goes here to minimize manually
 * converted code.
 */
public class ReplicationPlatform {

	@SuppressWarnings("unchecked")
	public static void copyCollectionState(Object original, Object destination, CounterpartFinder counterpartFinder) {
		Collection originalCollection = (Collection) original;
		Collection destinationCollection = (Collection) destination;
		destinationCollection.clear();
		Iterator it = originalCollection.iterator();
		while (it.hasNext()) {
			Object element = it.next();
			Object counterpart = counterpartFinder.findCounterpart(element);
			destinationCollection.add(counterpart);
		}
	}

	public static Collection emptyCollectionClone(Collection original) {
		if (original instanceof List) return new ArrayList(original.size());
		if (original instanceof Set) return new HashSet(original.size());
		return null;
	}
}

File: drs/drs.net/in/Db4objects/Db4o/Drs/Inside/ReplicationPlatform.cs
using System;
using Db4objects.Db4o.Reflect;

namespace Db4objects.Db4o.Drs.Inside
{
	/// <summary>
	/// Platform dependent code goes here to minimize manually
	/// converted code.
	/// </summary>
	public class ReplicationPlatform
	{
        public static void CopyCollectionState(object original, object destination, Db4objects.Db4o.Drs.Inside.ICounterpartFinder
			 counterpartFinder)
		{
			System.Collections.IList originalCollection = (System.Collections.IList
				)original;
			System.Collections.IList destinationCollection = (System.Collections.IList
				)destination;
			destinationCollection.Clear();
			System.Collections.IEnumerator it = originalCollection.GetEnumerator();
			while (it.MoveNext())
			{
				object element = it.Current;
				object counterpart = counterpartFinder.FindCounterpart(element);
				destinationCollection.Add(counterpart);
			}
		}

		public static System.Collections.ICollection EmptyCollectionClone(System.Collections.ICollection
			 original)
		{
			if (original is System.Collections.IList)
			{
				return new System.Collections.ArrayList(original.Count);
			}
			return null;
		}
	}
}

